Interop/Runtime/src/native/kotlin/kotlinx/cinterop/NativeTypes.kt

```diff
@@ -56,6 +56,59 @@ external fun CPointer<*>.getRawValue(): NativePtr
 
 internal fun CPointer<*>.cPointerToString() = "CPointer(raw=$rawValue)"
 
+
+public final class NativeVector private constructor() {
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getByte(index: Int): Byte
+
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getInt(index: Int): Int
+
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getLong(index: Int): Long
+
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getFloat(index: Int): Float
+
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getDouble(index: Int): Double
+
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getUByte(index: Int): UByte
+
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getUInt(index: Int): UInt
+
+    @TypedIntrinsic(IntrinsicType.EXTRACT_ELEMENT)
+    external fun getULong(index: Int): ULong
+
+    public override fun toString() = "(${getFloat(0)}, ${getFloat(1)}, ${getFloat(2)}, ${getFloat(3)})"
+
+    // Not as good for floating types
+    public fun equals(other: NativeVector): Boolean =
+            getLong(0) == other.getLong(0) && getLong(1) == other.getLong(1)
+
+    public override fun equals(other: Any?): Boolean =
+            other is NativeVector && this.equals(other)
+
+    override fun hashCode(): Int {
+        val x0 = getLong(0)
+        val x1 = getLong(1)
+        return 31 * (x0 xor (x0 shr 32)).toInt() + (x1 xor (x1 shr 32)).toInt()
```

Comment:
Replacing `(x0 xor (x0 shr 32)).toInt()` by `x0.hashCode()` (which is almost the same) would make this code simpler.